drivers/can/can_common.c

```diff
@@ -192,6 +192,8 @@ static int can_calc_timing_int(uint32_t core_clock, struct can_timing *res,
 	int sp_err;
 	struct can_timing tmp_res;
 
+	tmp_res.sjw = res->sjw;
```

Comment:
OK, I understand this a bit better now.  Why isn't the right fix:

    tmp_res = *res;

 so that the entire timing configuration that's updated in the loop has a defined state?

Reply:
Because the rest is written in the update function. No need for initializing. The SJW is not updated because it is not an outcome of the algorithm. SJW gets written in the drivers afterward. However, not changing the SJW makes sense in case someone sets it before the algorithm is called.

Reply:
Line 174-176

Reply:
Sure, I see that some fields are assigned in that, but without looking at the structure definition it's not clear that all of them are.  By assigning the entire structure you would reassure the reviewer/maintainer that *all* of it is at least as valid as the input, and that some future added field isn't going to get left uninitialized and also propagate to the caller.

Unless there's a clear motivation for not initializing the whole structure I'd really prefer this change.

Reply:
I tend to agree with @pabigot here. It will save us from encountering similar issues in the future if more fields are added to the struct.